File: II-2021/SOLUCIONARIO 1ER PARCIAL (LIC FELIPEZ)/2298: Aprendiendo a sumar.cpp
//https://jv.umsa.bo/problem.php?id=2298

#include <bits/stdc++.h>
 
using namespace std;
 
const int N = 1e7 + 5;
typedef long long ll;
int v[N];
ll pr[N];
int cur = 0;
void criba(){
    v[0] = v[1] = 1;
    for(int i = 2; i * i < N; i++){
        if(!v[i]){
            for(int j = i * i; j < N; j += i){
                v[j] = 1;
            }
        }
    }
 
    for(int i = 0; i < N; i++)
        if(!v[i]){
            pr[cur++] = (ll)i;
        }
 
    for(int i = 1; i < cur; i++)
        pr[i] += pr[i - 1];    
}
 
int main(){
 
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    criba();
 
    int q;
    cin >> q;
    while(q--){
        ll x;
        cin >> x;
        int a = 0, b = cur, mid;
        while(b - a > 1){
            mid = (a + b) >> 1;
            if(pr[mid] <= x)
                a = mid;
            else
                b = mid;
        }
 
        cout << a << '\n';
    }
 
    return 0;
 
}

File: II-2021/SOLUCIONARIO 1ER PARCIAL (LIC FELIPEZ)/2298: Aprendiendo a sumar.py
// https://jv.umsa.bo/problem.php?id=2298 

import sys
 
def criba():
    v[0] = v[1] = 0
    i = 2
    while i * i < N:
        if v[i] == 1:
            for j in range(i * i, N, i):
                v[j] = 0
        i += 1
 
    for i in range(2, N):
        if v[i] == 1:
            pr.append(i)
    for i in range(1, len(pr)):
        pr[i] += pr[i - 1]
 
N = int(1e7 + 5)
v = [1] * N
# guarda los números primos
pr = []    
 
criba()
 
 
t = int(sys.stdin.readline())
 
for _ in range(t):
    x = int(sys.stdin.readline())
 
    a, b, mid = 0, len(pr), 0
 
    while b - a > 1:
        mid = (a + b) >> 1
        if pr[mid] <= x:
            a = mid
        else:
            b = mid
    sys.stdout.write(str(a) + '\n')
